romancal/source_catalog/tests/test_source_catalog.py

```diff
@@ -432,3 +446,205 @@ def test_do_psf_photometry_column_names(tmp_path, image_model, fit_psf):
+    expected_result,
+    expected_outputs,
+    tmp_path,
+):
+    """
+    Test that the proper object is returned in the call to SourceCatalogStep
+    and that the desired output files are saved to the disk with the correct type.
+    """
+    os.chdir(tmp_path)
+    step = SourceCatalogStep()
+    result = step.call(
+        image_model,
+        bkg_boxsize=50,
+        kernel_fwhm=2.0,
+        snr_threshold=snr_threshold,
+        npixels=npixels,
+        save_results=save_results,
+        return_updated_model=return_updated_model,
+    )
+
+    # assert that we returned the correct object
+    assert isinstance(result, expected_result)
+
+    # assert that the desired output files were saved to disk
+    assert all(
+        Path(tmp_path / f"{result.meta.filename}_{suffix}.asdf").exists()
+        for suffix in expected_outputs.keys()
+    )
+
+    # assert that the desired output files were saved with the correct datamodel type
+    assert all(
+        isinstance(
+            rdm.open(Path(tmp_path / f"{result.meta.filename}_{suffix}.asdf")),
+            expected_outputs.get(suffix),
+        )
+        for suffix in expected_outputs.keys()
+    )
+
+
+@pytest.mark.webbpsf
+@pytest.mark.parametrize(
+    "snr_threshold, npixels, nsources, save_results, return_updated_model, expected_result, expected_outputs",
+    (
+        (
+            3,
+            10,
+            7,
+            True,
+            True,
+            MosaicModel,
+            {
+                "cat": MosaicSourceCatalogModel,
+                "segm": MosaicSegmentationMapModel,
+                "sourcecatalog": MosaicModel,
+            },
+        ),
+        (
+            3,
+            50,
+            5,
+            True,
+            False,
+            MosaicSourceCatalogModel,
+            {
+                "cat": MosaicSourceCatalogModel,
+                "segm": MosaicSegmentationMapModel,
+            },
+        ),
+        (
+            10,
+            10,
+            7,
+            False,
+            True,
+            MosaicModel,
+            {
+                "cat": MosaicSourceCatalogModel,
+                "segm": MosaicSegmentationMapModel,
+            },
+        ),
+        (
+            20,
+            10,
+            5,
+            False,
+            False,
+            MosaicSourceCatalogModel,
+            {
+                "cat": MosaicSourceCatalogModel,
+                "segm": MosaicSegmentationMapModel,
+            },
+        ),
+    ),
+)
+def test_l3_source_catalog_keywords(
+    mosaic_model,
+    snr_threshold,
+    npixels,
+    nsources,
+    save_results,
+    return_updated_model,
+    expected_result,
+    expected_outputs,
+    tmp_path,
+):
+    """
+    Test that the proper object is returned in the call to SourceCatalogStep
+    and that the desired output files are saved to the disk with the correct type.
+    """
+    os.chdir(tmp_path)
+    step = SourceCatalogStep()
```

Comment:
```suggestion
    step = SourceCatalogStep()
    # to mimic what happens in the elp set the "hidden" parameter
    # to cause this step to return a model instead of a catalog
    if return_updated_model:
        step.return_updated_model = True
```